go/vt/sqlparser/sql.y

```diff
@@ -5787,21 +5783,25 @@ UTC_DATE func_paren_opt
   {
     $$ = &JSONQuoteExpr{ StringArg:$3 }
   }
-| JSON_CONTAINS openb expression ',' expression json_path_param_list_opt closeb
+| JSON_CONTAINS openb expression ',' expression_list closeb
   {
-    $$ = &JSONContainsExpr{Target: $3, Candidate: $5, PathList: $6}
+    $$ = &JSONContainsExpr{Target: $3, Candidate: $5[0], PathList: $5[1:]}
   }
-| JSON_CONTAINS_PATH openb expression ',' expression ',' json_path_param_list closeb
+| JSON_CONTAINS_PATH openb expression ',' expression ',' expression_list closeb
   {
     $$ = &JSONContainsPathExpr{JSONDoc: $3, OneOrAll: $5, PathList: $7}
   }
-| JSON_EXTRACT openb expression ',' json_path_param_list closeb
+| JSON_EXTRACT openb expression ',' expression_list closeb
   {
     $$ = &JSONExtractExpr{JSONDoc: $3, PathList: $5}
   }
-| JSON_KEYS openb expression json_path_param_list_opt closeb
+| JSON_KEYS openb expression closeb
   {
-    $$ = &JSONKeysExpr{JSONDoc: $3, PathList: $4}
+    $$ = &JSONKeysExpr{JSONDoc: $3}
+  }
+| JSON_KEYS openb expression ',' expression closeb
```

Comment:
is this a bad way of doing it, @GuptaManan100? I have two cases here with `JSON_KEYS`

Reply:
No, this is fine as long as this happens only for one of these functions. In general I try to just reduce the number of total rules that are there in the grammar. So making `expression_comma_opt` will reduce one rule but add one more (its own definition), so it's not worth it.